schema/schema.go

```diff
@@ -38,6 +38,16 @@ func (s Schemas) GetByID(id string) (*Schema, error) {
 	return nil, errors.WithStack(herodot.ErrBadRequest.WithReasonf("Unable to find JSON Schema ID: %s", id))
 }
 
+func (s Schemas) GetIDs() ([]string, error) {
+	ids := []string{}
+
+	for _, ss := range s {
+		ids = append(ids, ss.ID)
+	}
+
+	return ids, nil
+}
```

Comment:
Instead of building a list of all schemas here and then slicing it later on, I think it makes more sense to paginate here directly and return what is needed.
```suggestion
func (s Schemas) List(page, perPage int) Schemas {
	ss := make(Schemas, 0, perPage)

	// TODO add guard checks
	copy(s[page*perPage, (page+1)*perPage], ss)

	return ss
}
```

Reply:
with the guard checks you mean checking that it doesn't exceed the boundaries of the slice that is copied from?

Reply:
Yes, exactly what you did already :wink: